test/DeathTest.cpp

```diff
@@ -144,7 +147,7 @@ TEST(DeathTest, ThreadSafeTest) {
       Death::Instance().RegisterDeathEvent(&DeathTest::RaceTest, "race");
    };
    std::vector<std::future<void>> waitingPromises;
-   for (int i = 0; i < 10000; i++) {
+   for (int i = 0; i < 10; i++) {
```

Comment:
Well. G3log has a way to register special handling in case of a fatal event (segfault et.al).

DeathKnell wraps that with its own API for triggering callbacks to whatever function you like in case of a fatal event.

Here we verify that that callback works on a cross-thread basis.